grails-app/services/com/unifina/service/StorageNodeService.groovy

```diff
@@ -41,15 +69,23 @@ class StorageNodeService {
 			thread.setUncaughtExceptionHandler(new NotifyStorageNodeTask.ErrorHandler())
 			thread.setName(String.format("AddStorageNodeTask[%s,%s]", streamId, storageNodeAddress))
 			thread.start()
-			return saved
+			return node
 		} else {
+			// TODO: https://linear.app/streamr/issue/BACK-155/assign-a-stream-to-a-storage-node-when-it-has-already-been-assigned
```

Comment:
In this case the method just return the existing `StreamStorageNode` (as BACK-155 states that nothing should happen if the stream has already been assigned)

Reply:
I'll fix it later. Frontend might need some notification about it.